core/src/main/java/org/pentaho/platform/web/http/api/resources/GenericFileResource.java

```diff
@@ -110,8 +100,32 @@ public IGenericFileTree getFolderSubtree( @NonNull @PathParam( "path" ) String b
     }
   }
 
+  private IGenericFileTree getTreeWithOptions( @Nullable String basePath,
+                                               @NonNull Integer maxDepth,
+                                               @NonNull String expandedPath,
+                                               @NonNull String filterString )
+    throws OperationFailedException {
+    GetTreeOptions options = new GetTreeOptions();
+
+    if ( basePath != null ) {
+      options.setBasePath( decodePath( basePath ) );
+    }
+    options.setMaxDepth( maxDepth );
+
+    try {
+      options.setFilter( filterString );
+    } catch ( IllegalArgumentException e ) {
+      throw new WebApplicationException( e, Response.Status.BAD_REQUEST );
```

Comment:
Should update the http status annotations and corresponding doclets, for both methods, to reflect the new case.